src/mem/impls.rs

```diff
@@ -70,7 +87,7 @@ pub unsafe fn copy_forward(mut dest: *mut u8, mut src: *const u8, mut n: usize)
         let mut prev_word = core::ptr::read_volatile(src_aligned);
 
         while dest_usize < dest_end {
-            src_aligned = src_aligned.add(1);
+            src_aligned = src_aligned.wrapping_add(1);
```

Comment:
FWIW this can actually be out-of-bounds arithmetic when `src_aligned` was rounded down, so there's another good reason to use `wrapping_add` here.